File: src/main/kotlin/by/mksn/kwitapi/configuration/security/CustomTokenEnhancer.java
package by.mksn.kwitapi.configuration.security;

import org.springframework.security.oauth2.common.OAuth2AccessToken;
import org.springframework.security.oauth2.provider.OAuth2Authentication;
import org.springframework.security.oauth2.provider.token.TokenEnhancer;
import org.springframework.stereotype.Component;

@Component
public class CustomTokenEnhancer implements TokenEnhancer {

    @Override
    public OAuth2AccessToken enhance(OAuth2AccessToken oAuth2AccessToken, OAuth2Authentication oAuth2Authentication) {
        return oAuth2AccessToken;
    }
}

File: src/main/kotlin/by/mksn/kwit/configuration/security/CustomTokenEnhancer.kt
package by.mksn.kwit.configuration.security

import org.springframework.security.oauth2.common.OAuth2AccessToken
import org.springframework.security.oauth2.provider.OAuth2Authentication
import org.springframework.security.oauth2.provider.token.TokenEnhancer
import org.springframework.stereotype.Component

@Component
open class CustomTokenEnhancer : TokenEnhancer {

    override fun enhance(
            oAuth2AccessToken: OAuth2AccessToken,
            oAuth2Authentication: OAuth2Authentication): OAuth2AccessToken
            = oAuth2AccessToken
}